gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -241,4 +402,11 @@ private void commitData()
       this.logger.error("Failed to update byte-level metrics of task " + this.taskId);
     }
   }
+
+  private void setForkState(ForkState expectedState, ForkState newState) {
```

Comment:
may need a better method name: it not only sets forkstate, but also sets taskstate, and does so only if current state = expectedstate